oci/archive/oci_transport.go

```diff
@@ -102,12 +67,13 @@ func NewReference(file, image string) (types.ImageReference, error) {
 	if err != nil {
 		return nil, err
 	}
-	// This is necessary to prevent directory paths returned by PolicyConfigurationNamespaces
-	// from being ambiguous with values of PolicyConfigurationIdentity.
-	if strings.Contains(resolved, ":") {
-		return nil, errors.Errorf("Invalid OCI reference %s:%s: path %s contains a colon", file, image, resolved)
+
+	err = internal.ValidateOCIPath(file)
+	if err != nil {
```

Comment:
(This could be `if err := … err != nil {`.  Applies to both transports.)

Reply:
sure. Personally, I don't like this style of the if statement, but fair enough